src/wp-includes/blocks/navigation.php

```diff
@@ -5,6 +5,648 @@
+		}
+
+		return $inner_blocks_html;
+	}
+
+	/**
+	 * Gets the inner blocks for the navigation block from the navigation post.
+	 *
+	 * @param array $attributes The block attributes.
+	 * @return WP_Block_List Returns the inner blocks for the navigation block.
+	 */
+	private static function get_inner_blocks_from_navigation_post( $attributes ) {
+		$navigation_post = get_post( $attributes['ref'] );
+		if ( ! isset( $navigation_post ) ) {
+			return new WP_Block_List( array(), $attributes );
+		}
+
+		// Only published posts are valid. If this is changed then a corresponding change
+		// must also be implemented in `use-navigation-menu.js`.
+		if ( 'publish' === $navigation_post->post_status ) {
+			$parsed_blocks = parse_blocks( $navigation_post->post_content );
+
+			// 'parse_blocks' includes a null block with '\n\n' as the content when
+			// it encounters whitespace. This code strips it.
+			$blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
+
+			if ( function_exists( 'get_hooked_blocks' ) ) {
+				// Run Block Hooks algorithm to inject hooked blocks.
+				$markup         = block_core_navigation_insert_hooked_blocks( $blocks, $navigation_post );
+				$root_nav_block = parse_blocks( $markup )[0];
+
+				$blocks = isset( $root_nav_block['innerBlocks'] ) ? $root_nav_block['innerBlocks'] : $blocks;
+			}
+
+			// TODO - this uses the full navigation block attributes for the
+			// context which could be refined.
+			return new WP_Block_List( $blocks, $attributes );
+		}
+	}
+
+	/**
+	 * Gets the inner blocks for the navigation block from the fallback.
+	 *
+	 * @param array $attributes The block attributes.
+	 * @return WP_Block_List Returns the inner blocks for the navigation block.
+	 */
+	private static function get_inner_blocks_from_fallback( $attributes ) {
+		$fallback_blocks = block_core_navigation_get_fallback_blocks();
+
+		// Fallback my have been filtered so do basic test for validity.
+		if ( empty( $fallback_blocks ) || ! is_array( $fallback_blocks ) ) {
+			return new WP_Block_List( array(), $attributes );
+		}
+
+		return new WP_Block_List( $fallback_blocks, $attributes );
+	}
+
+	/**
+	 * Gets the inner blocks for the navigation block.
+	 *
+	 * @param array    $attributes The block attributes.
+	 * @param WP_Block $block The parsed block.
+	 * @return WP_Block_List Returns the inner blocks for the navigation block.
+	 */
+	private static function get_inner_blocks( $attributes, $block ) {
+		$inner_blocks = $block->inner_blocks;
+
+		// Ensure that blocks saved with the legacy ref attribute name (navigationMenuId) continue to render.
+		if ( array_key_exists( 'navigationMenuId', $attributes ) ) {
+			$attributes['ref'] = $attributes['navigationMenuId'];
+		}
+
+		// If:
+		// - the gutenberg plugin is active
+		// - `__unstableLocation` is defined
+		// - we have menu items at the defined location
+		// - we don't have a relationship to a `wp_navigation` Post (via `ref`).
```

Comment:
```suggestion
		// - we don't have a relationship to a `wp_navigation` Post (via `ref`).
		// - we have menu items at the defined location
```
These two are in reverse order based on the code.

That said, the code appears to be self-explanatory. Is the comment needed?

Reply:
cc @scruffian

Reply:
This code has historically been complex and the source of bugs. These comments have existed for some time in the original code and have served us well in helping the contributors who work regularly with this code.